reconcile/openshift_saas_deploy_trigger_configs.py

```diff
@@ -23,27 +24,28 @@ def run(dry_run, thread_pool_size=10, internal=None,
         sys.exit(ExitCodes.ERROR)
     saas_files = [sf for sf in saas_files if is_in_shard(sf['name'])]
 
-    setup_options = {
-        'saas_files': saas_files,
-        'thread_pool_size': thread_pool_size,
-        'internal': internal,
-        'use_jump_host': use_jump_host,
-        'integration': QONTRACT_INTEGRATION,
-        'integration_version': QONTRACT_INTEGRATION_VERSION,
-    }
-    saasherder, jenkins_map, oc_map, settings = osdt_base.setup(setup_options)
+    saasherder, jenkins_map, oc_map, settings = \
+        osdt_base.setup(
+            saas_files=saas_files,
+            thread_pool_size=thread_pool_size,
+            internal=internal,
+            use_jump_host=use_jump_host,
+            integration=QONTRACT_INTEGRATION,
+            integration_version=QONTRACT_INTEGRATION_VERSION
+        )
     defer(lambda: oc_map.cleanup())
 
     trigger_specs = saasherder.get_configs_diff()
     # This will be populated by osdt_base.trigger in the below loop and
     # we need it to be consistent across all iterations
-    already_triggered = []
+    already_triggered = set()
 
-    error = False
-    for job_spec in trigger_specs:
-        trigger_error = osdt_base.trigger(
+    trigger_errors = \
+        threaded.run(
```

Comment:
Let's hold all this threading for now. If we want to run stuff concurrently, we can use `asyncio`, which is also far more readable.  Maybe its futures are easiest to understand and adapt for now. Multi-threading in Python is but an illusion, thanks to the GIL.

Let's have that conversation over slack.

Reply:
Our code is massively io-bound. Threading is working very well for us.